### ISSUE TITLE
Can't get player names in other languages

### ISSUE THREAD

**Issue by yumao233** [timhuijgen/ark-files#2]
Use: the latest version of ARKServer

hi,I used this project and it was very good.

But when the player name is Japanese, Chinese or other countries' language, the `CharacterName` property of the method `getPlayers()` is empty.
Tribe names have the same problem.

Cannot read languages of other countries. Is there any way to fix this problem? Or where to change the code?

**Comment by timhuijgen** [comment]
Hey, I'm glad the project is useful to you! I will have a look later today if I can fix this issue and get back to you once I know more.

**Comment by timhuijgen** [comment]
Hi, I've tested by inserting Chinese characters into the arkplayer file, but it works correctly in this use case. Can you share an Arkplayer file with me that does not work correctly so I can test/debug with that?

**Comment by yumao233** [comment]
Sorry for the late reply.

This is the arkprofile file and arktribe file on my ARK server:
download link: [Click here](https://drive.google.com/file/d/1t6X-m_12GmNJRenu7jdaWBvmm-kZMkNO/view?usp=sharing)

The detailed information is as follows:
The newly built ARK server has only one player, the player name is `人类` , and a tribe is created, all named in Chinese.

System environment:
Windows Server 2012 R2 Standard.
Nodejs 12.16.2.
ARK Server: latest version.


This is my `index.js`:
```javascript
const ArkFiles = require('ark-files');
let arkFiles = new ArkFiles('D:/ARK_ControlPanelPackage/server/server_core/TEST_Frist_ArkServer/');
let players = arkFiles.getPlayers();
let tribes = arkFiles.getTribes();
console.log(players);
console.log('=================================')
console.log(tribes);
```

Output:
```
D:\arkfs>node index.js
Fetching new player & tribe data from ark files
[
  {
    Tribe: {
      Players: [Array],
      Name: '',     //Failed to get name!
      OwnerId: 630427128,
      Id: false,
      TribeLogs: [Array],
      TribeMemberNames: [Array],
      FileCreated: '2020-04-21 05:25:55',
      FileUpdated: '2020-04-21 05:25:55'
    },
    PlayerName: '123',    //This is steam name
    Level: 1,
    TotalEngramPoints: false,
    CharacterName: '',     //Failed to get CharacterName!
    TribeId: false,
    SteamId: 76561198263639740, 
    PlayerId: 630427128,
    FileCreated: '2020-04-21 05:25:26',
    FileUpdated: '2020-04-21 05:26:21'
  }
]
=================================
[
  {
    Players: [ [Object] ],
    Name: '',      //Failed to get Name!
    OwnerId: 630427128,
    Id: false,
    TribeLogs: [ '' ],
    TribeMemberNames: [ '' ],
    FileCreated: '2020-04-21 05:25:55',
    FileUpdated: '2020-04-21 05:25:55'
  }
]
```

Hope this information can help you fix this problem. I really need this project, this project is very good.

Thank you very much!

**Comment by timhuijgen** [comment]
Hi Suwings, I've had a look but I don't have enough experience with Chinese encodings to fix this. The script is currently using utf8, but that isn't working for your file.

<img width="586" alt="Screenshot 2020-04-21 at 10 03 03" src="https://user-images.githubusercontent.com/11795446/79841362-975e1280-83b7-11ea-8c1a-658c6fe9c89e.png">
<img width="584" alt="Screenshot 2020-04-21 at 10 03 19" src="https://user-images.githubusercontent.com/11795446/79841366-988f3f80-83b7-11ea-88c8-fa4cb8b6f012.png">

Here you can see the difference in your file and one of mine. Maybe you can have a look yourself. This is where the property is read from the binary buffer: https://github.com/timhuijgen/ark-files/blob/master/src/ArkBinaryParser.js#L73

**Comment by yumao233** [comment]
In China, the Windows operating system uses `GBK` encoding by default.
In other words, if I change the Windows operating system to `UTF-8` encoding, can I solve this problem?

I will try this solution tonight and hope to solve it.

Thank you for helping me so much! Thanks again!

**Comment by timhuijgen** [comment]
That might work indeed :) good luck

**Comment by yumao233** [comment]
Hello! @timhuijgen .

I also feel very confused, I upgraded the system to Window Server 2019, and used UTF-8 as the system encoding, but the result still has not changed.

It is possible that this is not a problem of Windows, but a problem of ARK Server . I can't think of any other solutions, and I can't do anything about it.

I gave up, I can't solve this problem, and I don't know the reason. But there maybe a problem with ARKServer in Chinese environment.

Thank you for helping me before.

**Comment by timhuijgen** [comment]
@Suwings Did you try deleting (or temporarily removing) the arktribe and arkplayer files so that the Ark game will create new files in the utf-8 encoding?

**Comment by yumao233** [comment]
Of course, I deleted all the ark server files, re-installed the server using steamcmd, and let the player enter the server again, but the result is still the same.  I don't know why.
 
I think it maybe that arkserver recognizes that the operating system is still in Chinese, so there is still a problem with the file encoding.

Maybe I have a chance to solve this problem in the future, hope so.

thank you.

**Comment by yumao233** [comment]
Hi :)

Recently I tried some other tools, and finally I used this project [https://github.com/Qowyn/ark-tools](https://github.com/Qowyn/ark-tools) it can correctly convert Chinese, Japanese and other languages into json files, although a little troublesome , But can be converted correctly.

I would like to have a package that nodejs can use directly. Unfortunately, my experience is not enough to solve this problem. Maybe that project can help you more or less.

I just came to say thank you and your project.
